src/commands/new.rs

```diff
@@ -1,21 +1,37 @@
 use std::path::PathBuf;
-
-use clap::{Parser};
+use clap::{Parser, ArgGroup};
 use itertools::Itertools;
-
 use crate::hub_api;
 
 #[derive(Parser, Debug)]
 #[clap(about = "Create an application from a template on the Hub")]
+#[clap(group(
+    ArgGroup::new("operation")
+        .required(true)
+        .args(&["list", "name"])
+))]
 pub struct NewCommand {
     #[clap(short = 't')]
     terms: Vec<String>,
 
-    name: String,
+    #[clap(short, long)]
+    list: bool,
+
+    #[clap(name = "name", help = "Name of the application to create from the template")]
+    name: Option<String>,
 }
 
 impl NewCommand {
     pub async fn run(&self) -> anyhow::Result<()> {
+        if self.list {
+            return self.list_templates().await;
+        }
+
+        if self.name.is_none() {
+            println!("Please provide a name for the application you want to create.");
```

Comment:
Consider prompting rather than sending them back to reissue the command.

Also, in the Spin CLI, if the name is omitted, we prompt for the name _after_ they choose a template.  This saves entering a name and then realising that there's no suitable template.